src/main/java/org/broadinstitute/hellbender/engine/ReadWalker.java

```diff
@@ -131,4 +136,21 @@ protected final void onShutdown() {
         // Overridden only to make final so that concrete tool implementations don't override
         super.onShutdown();
     }
+
+    /**
+     * Create a common SAMFileWriter for use with ReadWalkers.
+     *
+     * @param outputFile    - if this file has a .cram extension then a reference is required. Can not be null.
+     * @param referenceFile - the reference source to use. Can not be null if a output file has a .cram extension.
+     * @param header        - header to be used for the output writer
+     * @param preSorted     - if true then the records must already be sorted to match the header sort order
+     * @return SAMFileWriter
+     */
+    protected SAMFileGATKReadWriter createCommonSAMWriter(
```

Comment:
They don't need to be the same, or override one another -- it's fine for the versions in `GATKTool` and `PicardCommandLineProgram` to return different types. Just wanted to make sure that when you rename one, you rename the other for some vague consistency across the codebase.